README.md

```diff
@@ -4,128 +4,336 @@ Daniel Ehrenberg, Igalia. Stage 3
+const mixed = [4n, 6, -12n, 10, 4, 0, 0n];
+// ↪  [4n, 6, -12n, 10, 4, 0, 0n]
+
+mixed.sort();
+// ↪ [-12n, 0, 0n, 10, 4n, 4, 6]
 ```
 
-Read and add two unsigned 64-bit integers in asm.js on the heap:
+### Conditionals
+
+A `BigInt` behaves like a `Number` in cases where it is converted to a `Boolean`: `if`, `||`, `&&`, `Boolean`, `!`.
 
 ```js
-function Add64Module(stdlib, foreign, buffer) {
-  "use asm";
-  var cast = stdlib.BigInt.asUintN;
-  var values = new stdlib.BigUint64Array(buffer);
-  function add64(aIndex, bIndex) {
-    aIndex = aIndex|0;
-    bIndex = bIndex|0;
-    var aValue = values[aIndex>>3];
-    var bValue = values[bIndex>>3];
-    return cast(64, aValue + bValue);
-  }
-  return { add64: add64 };
+
+if (0n) {
+  console.log('Hello from the if!');
+} else {
+	console.log('Hello from the else!');
 }
+
+// ↪ "Hello from the else!"
+
+0n || 12n
+// ↪ 12n
+
+0n && 12n
+// ↪ 0n
+
+Boolean(0n)
+// ↪ false
+
+Boolean(12n)
+// ↪ true
+
+!12n
+// ↪ false
+
+!0n
+// ↪ true
+
+```
+
+### Other API Notes
+
+`BigInt`s may also be used in `BigInt64Array` and `BigUint64Array` [typed arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray) for 64-bit integers.
+
+```js
+const view = new BigInt64Array(4);
+// ↪ [0n, 0n, 0n, 0n]
+view.length;
+// ↪ 4
+view[0];
+// ↪ 0n
+view[0] = 42n;
+view[0];
+// ↪ 42n
+
+// Highest possible BigInt value that can be represented as a
+// signed 64-bit integer.
+const max = 2n ** (64n - 1n) - 1n;
+view[0] = max;
+view[0];
+// ↪ 9_223_372_036_854_775_807n
+view[0] = max + 1n;
+view[0];
+// ↪ -9_223_372_036_854_775_808n
+//   ^ negative because of overflow
+
 ```
 
-## Use cases
+[For more about `BigInt` library functions, see the advanced section.](/ADVANCED.md)
+
+
+## Gotchas & Exceptions
 
-In many cases in JavaScript coding, integers larger than 2<sup>53</sup> come up, where casting to a double-precision float would lose real, relevant data:
-- Cases which would fit in a signed or unsigned 64-bit integer
-  - Reading certain machine registers, wire protocols
-  - Protobufs or JSON documents that have GUIDs in them
-  - `stat` may give some data as 64-bit integers
-  - Accurate timestamps
-- Bigger than 64-bit int cases
-  - Generally meeting a reasonable user expectation of a high-level language that integer arithmetic will be "correct" and not suddenly overflow
-  - Basis for implementing BigDecimal; perhaps some applications would be happy with fixpoint calculations based on a BigInt
-  - Any mathematical calculation with larger integers, e.g., solving Project Euler problems
-  - Exact geometric calculations
+### Interoperation with `Number` and `String`
 
-This proposal provides a new, second primitive numeric type, `BigInt`, to meet those use cases.
+The biggest surprise may be that `BigInt`s cannot be operated on interchangeably with `Number`s. Instead a `TypeError` will be thrown. ([Read the design philosophy for more about why this decision was made.]())
```

Comment:
Interesting that you identify this as a surprise. I guess I was defensively trying to be strictly positive about the proposal and *not* identify things as a surprise. What makes this behavior surprising? I wonder if we can phrase it in a way that makes people think, "that makes sense, of course it works that way" rather than "oh no, I'd better not trip up on this".

Reply:
I have to admit that this is probably the BigInt behavior I have the most trouble with,* based on my mental model of "how Javascript works". I understand conceptually why you chose the path of throwing an error rather than guessing which way to coerce an argument, especially in a situation where increased precision is the primary goal. But in a language where attempting to add two empty arrays or an array and an object doesn't throw an error even if if doesn't makes sense, doing so on two types that are closely related is, I think, a real departure and is going to surprise other folks.

All of which is to say, I do think calling it out as something you might trip on, especially for beginners, is very helpful.

What do you think about adding a bit here to explain more instead of hiding it behind a link, while also persisting in admitting that it may surprise you? Something like:

> It may be surprising to discover `BigInt`s cannot be operated on interchangeably with `Number`s. Instead a `TypeError` will be thrown. Why not just coerce them, like we can do with `Strings`? 

> In this case, coercion can be difficult to get right. If you write `2n ** 53n + 1.5`, what are you asking for? A float that is too big to be represented by a `Number` and will lose the precision you get from a `BigInt`? An integer where the `.5` is lopped off and the result is rounded down?

> Rather than guess wrong, `BigInt` will throw an error and ask you to be more clear about what you are looking for.


* No. 2 is is definitely the unary `+` for very similar reasons.